packages/web/src/pages/upload-page/fields/SourceFilesModalForm.tsx

```diff
@@ -0,0 +1,152 @@
+import { useCallback, useMemo } from 'react'
+
+import { StemCategory, StemUpload, removeNullable } from '@audius/common'
+import { Formik, useField } from 'formik'
+import { get, set } from 'lodash'
+
+import { ReactComponent as IconSourceFiles } from 'assets/img/iconSourceFiles.svg'
+import { Divider } from 'components/divider'
+
+import { processFiles } from '../store/utils/processFiles'
+
+import { ModalField } from './ModalField'
+import styles from './SourceFilesModalForm.module.css'
+import {
+  SourceFilesView,
+  dropdownRows as stemCategories
+} from './SourceFilesView'
+import { ToggleRowField } from './ToggleRowField'
+
+const ALLOW_DOWNLOAD = 'download.is_downloadable'
+const FOLLOWER_GATED = 'download.requires_follow'
+const STEMS = 'stems'
+
+const messages = {
+  title: 'Stems & Source Files',
+  description:
+    'Upload your stems and source files to help your fans remix your tracks!',
+  previewDescription:
+    'Upload stems and source files for your music, enabling others to create remixes with ease.',
+  [ALLOW_DOWNLOAD]: {
+    header: 'Make Full MP3 Track Available',
+    description:
+      'Enable to provide your fans with a copy of your track as an mp3 file'
+  },
+  [FOLLOWER_GATED]: {
+    header: 'Available Only to Followers',
+    description:
+      'Make your stems and source files available only to your followers'
+  }
+}
+
+export type SourceFilesFormValues = {
+  [ALLOW_DOWNLOAD]: boolean
+  [FOLLOWER_GATED]: boolean
+  [STEMS]: StemUpload[]
+}
+
+export const SourceFilesModalForm = () => {
+  // These refer to the field in the outer EditForm
+  const [{ value: allowDownloadValue }, , { setValue: setAllowDownloadValue }] =
+    useField(ALLOW_DOWNLOAD)
+  const [{ value: followerGatedValue }, , { setValue: setFollowerGatedValue }] =
+    useField(FOLLOWER_GATED)
+  // TODO: Stems goes outside tracks in uploadTracks
+  const [{ value: stemsValue }, , { setValue: setStemsValue }] = useField(STEMS)
+
+  const initialValues = useMemo(() => {
+    const initialValues = {}
+    set(initialValues, ALLOW_DOWNLOAD, allowDownloadValue)
+    set(initialValues, FOLLOWER_GATED, followerGatedValue)
+    set(initialValues, STEMS, stemsValue ?? [])
+    return initialValues as SourceFilesFormValues
+  }, [allowDownloadValue, followerGatedValue, stemsValue])
+
+  const onSubmit = (values: SourceFilesFormValues) => {
```

Comment:
if this is new every time there is no point in memoizing the initial values, there is a chance we can just not worry about the memo, up to you